x-pack/plugins/alerting/server/lib/monitoring.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import stats from 'stats-lite';
+import { RuleMonitoring } from '../types';
+
+export const getExecutionSuccessRatio = (ruleMonitoring: RuleMonitoring) => {
+  const { history } = ruleMonitoring.execution;
+  return history.filter(({ success }) => success).length / history.length;
+};
+
+export const getExecutionDurationPercentiles = (ruleMonitoring: RuleMonitoring) => {
+  const durationSamples = ruleMonitoring.execution.history.reduce((duration, history) => {
```

Comment:
nit -> to avoid the cast in L22
```suggestion
  const durationSamples = ruleMonitoring.execution.history.reduce<number[]>((duration, history) => {
```